### PR TITLE
applepay payment + giftcard compatibility

### PR DESCRIPTION
**Pull request by descorp**
# Open PR

## Changes

* update integration docs
* fix missed payment for ApplePayComponent
* fix gift-card flow compatibility for ApplePay on DropIn

### CODE REVIEW

File: AdyenComponents/Apple Pay/ApplePayConfiguration.swift [hunk 1]
```diff
@@ -100,6 +100,14 @@ extension ApplePayComponent {
 
             return paymentRequest
         }
+
+        @_spi(AdyenInternal)
+        public func updated(amount: Amount, localeIdentifier: String?) -> Self {
```
Comment (erenbesel) [inline AdyenComponents/Apple Pay/ApplePayConfiguration.swift]: Maybe `byUpdating(amount: .....)` more apple type of naming

Comment (descorp) [reply AdyenComponents/Apple Pay/ApplePayConfiguration.swift]: It returns new modified instance of struct, just like `.sorted()`, `.capitalized()` or `.uppercased()` 🤔

File: AdyenDropIn/Utilities/ComponentManager.swift [hunk 2]
```diff
@@ -176,10 +176,21 @@ internal final class ComponentManager {
...
+
+        if let amount = order?.remainingAmount ?? remainingAmount {
+            let configuration = applePay.updated(amount: amount, localeIdentifier: configuration.localizationParameters?.locale)
+            if let component = try? PreApplePayComponent(paymentMethod: paymentMethod,
+                                                         context: context,
+                                                         configuration: preApplePayConfig,
+                                                         applePayConfiguration: configuration) {
+                return component
+            }
+        }
```
Comment (mohammedDehairy) [inline AdyenDropIn/Utilities/ComponentManager.swift:193]:
```suggestion
        if let amount = order?.remainingAmount ?? remainingAmount,
         let configuration = applePay.updated(amount: amount, localeIdentifier: configuration.localizationParameters?.locale),
         let component = try? PreApplePayComponent(paymentMethod: paymentMethod,
                                                          context: context,
                                                          configuration: preApplePayConfig,
                                                          applePayConfiguration: configuration)
          {
            
            return component
         }
```

File: AdyenComponents/Apple Pay/ApplePayPayment.swift [hunk 3]
```diff
@@ -14,10 +14,10 @@ import PassKit
 public struct ApplePayPayment {
 
     /// The amount for this payment.
-    public let payment: Payment
+    public private(set) var payment: Payment
```
Comment (mohammedDehairy) [inline AdyenComponents/Apple Pay/ApplePayPayment.swift:17]: what about the other `Payment?` object, should we do as we did in v4?

File: AdyenComponents/Apple Pay/ApplePayConfiguration.swift (hunk 1 above)
Comment (descorp) [reply AdyenComponents/Apple Pay/ApplePayConfiguration.swift]: Agreed on `updating(amount: localeIdentifier)`

File: AdyenDropIn/Utilities/ComponentManager.swift (hunk 2 above)
Comment (descorp) [reply AdyenDropIn/Utilities/ComponentManager.swift:193]: `applePay.updated(amount: amount, localeIdentifier: configuration.localizationParameters?.locale)` in non-optional, so you can't chain them in IF 😔

File: AdyenComponents/Apple Pay/ApplePayPayment.swift (hunk 3 above)
Comment (descorp) [reply AdyenComponents/Apple Pay/ApplePayPayment.swift:17]: This seems like old version 🤔
It is exactly like v4 [right now](https://github.com/Adyen/adyen-ios/pull/934/files#diff-d77f02f47479a235fefb85a4f13afa8284eecc0df365f3b01af78d4d86e8aed3R31-R34)

**Review by erenbesel — APPROVED** [review]

**Review by mohammedDehairy — APPROVED** [review]

**Pull request merged**